### PR TITLE
Stats improvements

### PR DESCRIPTION
**Pull request by richmolj**

### CODE REVIEW

**Comment by richmolj** [comment]
@wadetandy

File: lib/graphiti/stats/payload.rb [hunk 1]
```diff
@@ -31,7 +32,9 @@ def generate
 
             each_calculation(name, calculation) do |calc, function|
               args = [@scope, name]
-              args << @resource.context if function.arity == 3
+              args << @resource.context if function.arity >= 3
```
Comment (wadetandy) [inline lib/graphiti/stats/payload.rb]: Starting to get a bit options heavy for my liking.  How about creating a class to hold these members and pass that in instead?

File: spec/stats_spec.rb [hunk 2]
```diff
@@ -142,6 +142,36 @@
             .to eq({ 'age' => { 'second' => 1338 } })
         end
       end
+
+      context 'that resolves from response data' do
+        class Results < Array
```
Comment (wadetandy) [inline spec/stats_spec.rb]: I know this is just for the sake of tests so I don't care if this stays here, but I think in the case of documentation, I'd prefer examples use decorators instead:

```ruby
class Results < SimpleDelegator
  attr_accessor :meta
  
  def initialize(array, meta:)
    super(array)
    @meta = meta
  end
end
```

This way we don't have to recreate the array every time, which should improve performance.

**Review by wadetandy — COMMENTED** [review]
A couple comments but overall nice change.

File: spec/stats_spec.rb (hunk 2 above)
Comment (richmolj) [reply spec/stats_spec.rb]: Good call; updated the test

File: lib/graphiti/stats/payload.rb (hunk 1 above)
Comment (richmolj) [reply lib/graphiti/stats/payload.rb]: Not against it for a future PR if we find ourselves hitting this class again, but going to leave it for now.

**Pull request merged**